deel/lip/activations.py

```diff
@@ -126,20 +126,22 @@ def build(self, input_shape):
             self.n = input_shape[self.channel_axis]
         if (input_shape[self.channel_axis] % self.n) != 0:
             raise RuntimeError("self.n has to be a divisor of the number of channels")
+        self.flat_shape = [-1, tf.reduce_prod(input_shape[1:]).numpy() // self.n, self.n]
+        self.out_shape = tuple((-1,)) + tuple(input_shape.as_list()[1:])
```

Comment:
To handle different `input_shape` types (e.g. `build((32,32,3))` or `build([32,32,3])`), it could be better to ensure that input_shape here is a TensorShape object:
`input_shape = tf.TensorShape(input_shape)`

Moreover, line 130 could be simplified:
`self.out_shape = (-1,) + tuple(input_shape.as_list()[1:])`
or
`self.out_shape = [-1] + input_shape.as_list()[1:]`

Reply:
your're right about `input_shape = tf.TensorShape(input_shape)` which could ensure the that the input_shape has a `as_list` method